packages/graph-explorer-proxy-server/node-server.js

```diff
@@ -174,16 +174,59 @@ async function fetchData(res, next, url, options, isIamEnabled, region, serviceT
   }
   // POST endpoint for SPARQL queries.
   app.post("/sparql", async (req, res, next) => {
+    // Gather info from the headers
+    const queryId = req.headers["queryid"];
+    const graphDbConnectionUrl = req.headers["graph-db-connection-url"];
+    const isIamEnabled = !!req.headers["aws-neptune-region"];
+    const region = isIamEnabled ? req.headers["aws-neptune-region"] : "";
+    const serviceType = isIamEnabled ? (req.headers["service-type"] ?? DEFAULT_SERVICE_TYPE) : "";
+
+    /// Function to cancel long running queries if the client disappears before completion
+    async function cancelQuery() {
+      if (!queryId) {
+        return;
+      }
+      proxyLogger.debug(`Cancelling request ${queryId}...`);
+      return await retryFetch(
+        new URL(`${graphDbConnectionUrl}/sparql/status`),
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/x-www-form-urlencoded",
+          },
+          body: `cancelQuery&queryId=${queryId}&silent=true`,
```

Comment:
Should we validate `queryId` to make sure other parameters can not be injected into the body?
Reference: https://owasp.org/Top10/A03_2021-Injection/